packages/contracts/src/contracts/current/protocol/Exchange/MixinSignatureValidator.sol

```diff
@@ -69,14 +82,18 @@ contract MixinSignatureValidator is MSignatureValidator {
         // it an explicit option. This aids testing and analysis. It is
         // also the initialization value for the enum type.
         if (signatureType == SignatureType.Illegal) {
-            revert();
+            // NOTE: Reason cannot be assigned to a variable because of https://github.com/ethereum/solidity/issues/4051
+            revert("Illegal signature type.");
```

Comment:
could you `require(false,` to get around this?

Reply:
Probably, I'm fine just waiting until the issue is fixed. The TODO is a good reminder that we need to fix this either way.